granulate_utils/linux/ns.py

```diff
@@ -67,6 +68,53 @@ def resolve_proc_root_links(proc_root: str, ns_path: str) -> str:
     return path
 
 
+def get_process_nspid(pid: int) -> Optional[int]:
+    with open(f"/proc/{pid}/status") as f:
+        for line in f:
+            fields = line.split()
+            if fields[0] == "NSpid:":
+                return int(fields[-1])
+
+    if is_same_ns(pid, NsType.pid.name):
+        # If we're in the same PID namespace, then the outer PID is also the inner pid (NSpid)
+        return pid
+
+    return _get_process_nspid_by_sched_files(pid)
+
+
+def _get_process_nspid_by_sched_files(pid: int):
+    # Old kernel (pre 4.1) doesn't have an NSpid field in their /proc/pid/status file
+    # Instead, we can look through all /proc/*/sched files from inside the process' pid namespace, and due to a bug
+    # (fixed in 4.14) the outer PID is exposed, so we can find the target process by comparing the outer PID
+
+    def _find_inner_pid() -> Optional[int]:
+        pattern = re.compile(r"\((\d+), #threads: ")
```

Comment:
Give an example to how one such line looks like? (always a good practice when writing regex - so the reader can see what you're trying to match)